### PR TITLE
Feat: Parse events

### PR DESCRIPTION
**Pull request by tmrdlt**
- Parse relevant events based on `log.event_hash` and known event hashes (same for testnet and mainnet)
- Get `reserve0`, `reserve1`, `deltaReserve0`, `deltaReserve1` from log.data or log.args
- Merge Sync events with succeeding non-Sync events
- Fixed BigInts in database (use psql `Decimal`)
- Fixed sorting of events by utilizing `transactionIndex`
- Added `@typescript-eslint/strict-boolean-expressions` rule and fixed problems

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -77,36 +77,38 @@ model PairLiquidityInfoHistoryError {
...
 
 model PairLiquidityInfoHistoryV2 {
+  id               Int      @id @default(autoincrement())
+  pair             Pair     @relation(fields: [pairId], references: [id])
+  pairId           Int
+  eventType        String
+  reserve0         Decimal  @db.Decimal(100, 0)
+  reserve1         Decimal  @db.Decimal(100, 0)
+  deltaReserve0    Decimal? @db.Decimal(100, 0) // Null if event is a manually triggered Sync event
+  deltaReserve1    Decimal? @db.Decimal(100, 0) // Null if event is a manually triggered Sync event
```
Comment (kenodressel) [inline prisma/schema.prisma]: we could also make it not optional and use `0` or calculate even calculate it?

File: src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts [hunk 2]
```diff
@@ -92,63 +119,115 @@ export class PairLiquidityInfoHistoryImporterV2Service {
...
+        const logsAndEvents = relevantContractLogs
+          .map((log) => {
+            const event = this.parseEvent(log);
+            if (event) {
+              return {
+                log,
+                event,
+              };
+            }
+          })
```
Comment (kenodressel) [inline src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts]:
```suggestion
          .map((log) => ({
             log,
             event: this.parseEvent(log)
            }))
```

File: src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts [hunk 3]
```diff
@@ -92,63 +119,115 @@ export class PairLiquidityInfoHistoryImporterV2Service {
...
+          .filter(
+            (logAndEvent): logAndEvent is { log: ContractLog; event: Event } =>
+              !!logAndEvent,
```
Comment (kenodressel) [inline src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts]:
```suggestion
            (logAndEvent): logAndEvent is { log: ContractLog; event: Event } =>
              !!logAndEvent.event,
```

Comment (kenodressel) [reply src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts]: Do we need the `is` part still, did you try to get rid of it?

File: src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts (hunk 2 above)
Comment (kenodressel) [reply src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts]: Could also be extended with ` as { log: ContractLog; event?: Event }`

File: src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts [hunk 4]
```diff
@@ -10,6 +10,22 @@ import { ContractAddress } from '../../clients/sdk-client.model';
...
+  PairBurn = 'PairBurn',
+}
+
+type Event = {
+  eventType: EventType;
+  reserve0?: bigint;
+  reserve1?: bigint;
+  deltaReserve0?: bigint;
+  deltaReserve1?: bigint;
+};
```
Comment (kenodressel) [inline src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts]: should we create 4 events here? eg for sync we will never get a deltaReserve and for SwapTokens never a reserve

File: src/tasks/tasks.service.ts [hunk 5]
```diff
@@ -38,6 +40,23 @@ export class TasksService {
     }
   }
 
+  // TODO set correct frequency
+  @Cron(CronExpression.EVERY_10_SECONDS)
```
Comment (kenodressel) [inline src/tasks/tasks.service.ts]: should we take it in like this?

File: prisma/schema.prisma (hunk 1 above)
Comment (tmrdlt) [reply prisma/schema.prisma]: I'd say `0` makes sense, as during a sole sync event, there is nothing added or removed, what do you think?

File: src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts (hunk 3 above)
Comment (tmrdlt) [reply src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts]: I tried without it and got errors during compilation :/

File: src/tasks/tasks.service.ts (hunk 5 above)
Comment (tmrdlt) [reply src/tasks/tasks.service.ts]: Probably not. I'd just run the v2 instead of the v1 I think!

File: prisma/schema.prisma (hunk 1 above)
Comment (kenodressel) [reply prisma/schema.prisma]: Sync events are actually triggered to update the balance from the token contract to the pair contract, so if somebody were to send some tokens to the pair, the reserves in the pair would not be updated until a sync is triggered.

Comment (tmrdlt) [reply prisma/schema.prisma]: Good point, so then it would make more sense calculating it IMO.

**Review by kenodressel — APPROVED** [review]

**Pull request merged**